pkg/platform/baremetal/baremetal.go

```diff
@@ -64,7 +63,12 @@ func (c *config) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext)
 	if configBody == nil {
 		return hcl.Diagnostics{}
 	}
-	return gohcl.DecodeBody(*configBody, evalContext, c)
+
+	if diags := gohcl.DecodeBody(*configBody, evalContext, c); len(diags) != 0 {
```

Comment:
We should be using `diags.HasErrors()` instead of `len` comparison, to use the provided API by the diagnostics. This will mean less work if we decide to also produce warnings from the configuration. I also think it is easier to read.